low-level/files/opt/seagate/sspl/setup/resource_map/lr2/storage.py

```diff
@@ -109,6 +109,36 @@ def get_health_info(self, rpath):
                 errno.EINVAL, msg)
         return info
 
+    @staticmethod
+    def get_fan_specfics(fan):
+        return {
+            "uid": fan.get("durable-id", "NA"),
+            "location": fan.get("location", "NA"),
+            "status": fan.get("status", "NA"),
+            "speed": fan.get("speed", "NA"),
+            "serial-number": fan.get("serial-number", "N/A"),
+            "part-number": fan.get("part-number", "N/A"),
+            "health": fan.get("health", "OK"),
+        }
+
+    def get_fanmodules_info(self):
+        response = []
+        fanmoduels_data = self.get_realstor_encl_data("fan-modules")
+        if fanmoduels_data:
+            for fan_module in fanmoduels_data:
+                uid = fan_module.get('durable-id', 'NA')
+                health = fan_module.get('health')
+                fan_module_resp = self.get_health_template(uid, is_fru=True)
+                specifics = [self.get_fan_specfics(fan) for fan in fan_module['fan']]
+                description = f"FAN is in {'good' if health=='OK' else 'bad'} health"
+                recommendation = fan_module.get('health-recommendation', 'NA'),
```

Comment:
set_health_data func will take care of description & recommendation fields based on status.
No need to explicitly pass these params

Reply:
Done.

Reply:
can remove #L133-134, not getting used right?

Reply:
yeah. removed.